### PR TITLE
Hidden wallet number

### LINKED ISSUE #2467: Trezor Suite - hidden wallet - number missing
image

image

### LINKED ISSUE #2578: Hidden wallet - Standard wallet seemingly included with the hidden wallet numbers
> Suite: app 20.10.1 a263552
> OS: Windows 10
> Device: model T 2.3.4 50854b921
 
**Description**
Misleading numbering when a Standard, non-passphrase, wallet is added. 

**Steps to reproduce:**
**A)**
1. Add a Standard wallet
2. Add a Hidden wallet
3. Observe

**B)**
1. Add a Hidden wallet
2. Add a Standard wallet
3. Observe

**Expected result:**
Standard wallet is not seemingly included in how Hidden wallets are numbered.

**Actual result:**
A) Hidden wallet is numbered # 2 and so on when Standard wallet is added first,
B) the Hidden wallets are numbered correctly until a Standard wallet is added; 1 number is skipped and continued from there.

**Screenshots:**
 Standard wallet added **first** | Standard wallet added **later**
 -|- 
image | image

### PR DESCRIPTION
**Pull request by mroz22**
Goals here: 
- fix #2467
- fix #2578

todo:
- [x] migration
- [x] test migration
- [x] squash
- [x] remove some comments

### CODE REVIEW

File: packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/components/AddWalletButton/index.tsx [hunk 1]
```diff
@@ -39,7 +39,11 @@ interface Props {
 
 const AddWalletButton = ({ device, instances, addDeviceInstance, selectDeviceInstance }: Props) => {
     const hasAtLeastOneWallet = instances.find(d => d.state);
-    const emptyPassphraseWalletExists = instances.find(d => d.useEmptyPassphrase); // Find a "standard wallet" among user's wallet instances. If no such wallet is found, the variable is undefined.
+    // Find a "standard wallet" among user's wallet instances. If no such wallet is found, the variable is undefined.
+    const emptyPassphraseWalletExists = instances.find(
+        // todo: this change might to make sense
+        d => (d.useEmptyPassphrase || !d.instance) && d.state,
```
Comment (mroz22) [inline packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/components/AddWalletButton/index.tsx]: remove instance

File: packages/suite/src/components/suite/SwitchDevice/components/WalletInstance/Container.tsx [hunk 2]
```diff
@@ -32,6 +32,7 @@ interface OwnProps {
     enabled: boolean;
     selected: boolean;
     selectDeviceInstance: (instance: TrezorDevice) => void;
+    index: number;
```
Comment (mroz22) [inline packages/suite/src/components/suite/SwitchDevice/components/WalletInstance/Container.tsx]: comment

File: packages/suite/src/components/suite/SwitchDevice/components/WalletInstance/index.tsx [hunk 3]
```diff
@@ -115,9 +111,9 @@ const WalletInstance = ({
                                 payload={{
                                     type: 'walletLabel',
                                     deviceState: instance.state,
-                                    defaultValue: !instance.instance
+                                    defaultValue: instance.useEmptyPassphrase
```
Comment (mroz22) [inline packages/suite/src/components/suite/SwitchDevice/components/WalletInstance/index.tsx:114]: ?

File: packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/components/AddWalletButton/index.tsx (hunk 1 above)
Comment (mroz22) [reply packages/suite/src/components/suite/SwitchDevice/components/DeviceItem/components/AddWalletButton/index.tsx]: fixed

File: packages/suite/src/components/suite/SwitchDevice/components/WalletInstance/index.tsx (hunk 3 above)
Comment (mroz22) [reply packages/suite/src/components/suite/SwitchDevice/components/WalletInstance/index.tsx:114]: should be ok. is not used as a key.

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**